auth/realm/base/src/main/java/org/wildfly/security/auth/realm/DistributedSecurityRealm.java

```diff
@@ -0,0 +1,267 @@
+            if (!currentIdentity.exists()) {
+                return nextIdentity();
+            }
+            return true;
+        }
+
+        @Override
+        public Principal getRealmIdentityPrincipal() {
+            return currentIdentity.getRealmIdentityPrincipal();
+        }
+
+        @Override
+        public SupportLevel getCredentialAcquireSupport(final Class<? extends Credential> credentialType, final String algorithmName, final AlgorithmParameterSpec parameterSpec) throws RealmUnavailableException {
+            return currentIdentity.getCredentialAcquireSupport(credentialType, algorithmName, parameterSpec);
+        }
+
+        @Override
+        public SupportLevel getEvidenceVerifySupport(final Class<? extends Evidence> evidenceType, final String algorithmName) throws RealmUnavailableException {
+            return currentIdentity.getEvidenceVerifySupport(evidenceType, algorithmName);
+        }
+
+        @Override
+        public <C extends Credential> C getCredential(final Class<C> credentialType, final String algorithmName, final AlgorithmParameterSpec parameterSpec) throws RealmUnavailableException {
+            do {
+                C credential = currentIdentity.getCredential(credentialType, algorithmName, parameterSpec);
+                if (credential != null) {
+                    return credential;
+                }
+            } while (nextIdentity());
+
+            return null;
+        }
+
+        @Override
+        public <C extends Credential> C getCredential(final Class<C> credentialType, final String algorithmName) throws RealmUnavailableException {
+            do {
+                C credential = currentIdentity.getCredential(credentialType, algorithmName);
+                if (credential != null) {
+                    return credential;
+                }
+            } while (nextIdentity());
+
+            return null;
+        }
+
+        @Override
+        public <C extends Credential> C getCredential(final Class<C> credentialType) throws RealmUnavailableException {
+            do {
+                C credential = currentIdentity.getCredential(credentialType);
+                if (credential != null) {
+                    return credential;
+                }
+            } while (nextIdentity());
+
+            return null;
+        }
+
+        @Override
+        public boolean verifyEvidence(final Evidence evidence) throws RealmUnavailableException {
+            return currentIdentity.verifyEvidence(evidence);
+        }
+
+        @Override
+        public boolean exists() throws RealmUnavailableException {
+            return currentIdentity.exists();
+        }
+
+        @Override
+        public AuthorizationIdentity getAuthorizationIdentity() throws RealmUnavailableException {
+            return currentIdentity.getAuthorizationIdentity();
+        }
+
+        @Override
+        public void dispose() {
+            for (RealmIdentity identity : loadedIdentities) {
```

Comment:
It doesn't look like `loadedIdentities` actually gets populated so it looks like this block should be removed and replaced with `currentIdentity.dispose()`.

Reply:
Will fix this in a follow-up PR.

Reply:
Fixed this in https://github.com/wildfly-security/wildfly-elytron/pull/1426